### PR TITLE
VertexLoaderX64: pre-allocate code space

### PR DESCRIPTION
**Pull request by Tilka**
Should partially fix [issue 8180](https://code.google.com/p/dolphin-emu/issues/detail?id=8180).

This is just to show what I had in mind. The code should be moved elsewhere before merging.

~~TODO: ARM.~~

### CODE REVIEW

**Comment by Sonicadvance1** [comment]
ARM doesn't need this nonsense, it already allocates everything without issue.
Not like our x86 code where it requires everything in lowmem.

**Comment by magumagu** [comment]
Is there some reason vertex loaders need to be page-aligned?  You could just reuse the emitter rather than explicitly allocating a page for each vertex loader.

**Comment by Tilka** [comment]
I have no idea why that hasn't occurred to me.

File: Source/Core/VideoCommon/VertexLoaderX64.cpp [hunk 1]
```diff
@@ -16,15 +16,25 @@ static const X64Reg scratch3 = ABI_PARAM4;
...
 
+static class VertexLoaderCode : public Gen::X64CodeBlock
+{
+public:
+	VertexLoaderCode(size_t size)
+	{
+		AllocCodeSpace(size);
+		ClearCodeSpace();
+	}
+} s_vertex_loader_code(2*1024*1024);
```
Comment (delroth) [inline Source/Core/VideoCommon/VertexLoaderX64.cpp]: How many games start/stop before this overflows? :)

Comment (Tilka) [inline Source/Core/VideoCommon/VertexLoaderX64.cpp]: My eyes can barely focus on what I'm typing, so I blame lack of sleep.

File: Source/Core/VideoCommon/VertexLoaderX64.cpp [hunk 2]
```diff
@@ -16,15 +17,33 @@ static const X64Reg scratch3 = ABI_PARAM4;
...
+		ClearCodeSpace();
+	}
+};
+
+static std::unique_ptr<VertexLoaderCode> s_vertex_loader_code;
+
+
+void VertexLoaderX64::Initialize()
+{
+	s_vertex_loader_code = std::make_unique<VertexLoaderCode>(2 * 1024 * 1024);
```
Comment (FioraAeterna) [inline Source/Core/VideoCommon/VertexLoaderX64.cpp]: should this be a constant or define (to match similar cases in the JIT?)

**Comment by FioraAeterna** [comment]
IMO this needs recover code in case we run out of space, i.e. to clear all the loaders and start over (unless it does and I'm missing it)

**Comment by Tilka** [comment]
Superseeded by #2487.

**Pull request closed without merging**